### PR TITLE
LPD-35448 - DSM Filter edition works 

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Story / Task
[LPD-26397](https://liferay.atlassian.net/browse/LPD-26397) / [LPD-35448](https://liferay.atlassian.net/browse/LPD-35448)

## Goal
Update edition process to ensure that changes are applied for filters with and without the feature flag enabled.

## PW
image

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/components/Configuration.tsx [hunk 1]
```diff
@@ -119,7 +115,8 @@ function Configuration({
 					}: {
 						selectedFields: Array<IField>;
 					}) => {
-						setSelectedField(selectedFields[0]);
+
+						// setSelectedField(selectedFields[0]);
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/components/Configuration.tsx:119]: :broom:

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/components/Configuration.tsx [hunk 2]
```diff
@@ -216,7 +214,8 @@ function Configuration({
 							});
 
 							if (newVal) {
-								setSelectedField(newVal);
+
+								// setSelectedField(newVal);
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/components/Configuration.tsx:218]: :broom:

**Review by markocikos — COMMENTED** [review]
@juanjofgliferay looks good, some minor bits inline

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx [hunk 3]
```diff
@@ -85,10 +85,10 @@ type FilterCollection = Array<IFilter>;
 interface IPropsFilterFormComponent {
 	dataSet: IDataSet | FDSViewType;
 	fdsFilterClientExtensions?: IClientExtensionRenderer[];
-	fieldNames?: string[];
 	fields: IField[];
 	filter?: IFilter | ISelectionFilter;
 	filterType?: EFilterType;
+	inUseFieldNames?: string[];
```
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:91]: Those fields were used :smile:  to display the `IN USE` tag in the dropdown and validate the form control against created filters. 
`fieldNames` sounded more like all the field names in the dropdown/tree, even if not used.

Alias sounds like a good option to clarify the scope

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! :sparkles:

**Comment by juanjofgliferay** [comment]
ci:forward

**Pull request closed without merging**